internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -54,14 +96,114 @@ func TestDispatchThrottlingCheckResolver(t *testing.T) {
 		require.False(t, req.GetRequestMetadata().WasThrottled.Load())
 	})
 
+	t.Run("dispatch_should_use_request_threshold_if_available", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
+			// We set timer ticker to 1 hour to avoid it interfering with test
+			Frequency:        1 * time.Hour,
+			DefaultThreshold: 200,
+			MaxThreshold:     210,
+		}
+		dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
+		defer dut.Close()
+
+		initialMockResolver := NewMockCheckResolver(ctrl)
+		dut.SetDelegate(initialMockResolver)
+
+		// this is to simulate how many times request has been dispatched
+		// Since this is a small value, we will not expect throttling
+		req := &ResolveCheckRequest{RequestMetadata: NewCheckRequestMetadata(10)}
+		req.GetRequestMetadata().DispatchCounter.Store(203)
```

Comment:
The threshold is 205 below, but this is 203. If we want this to be throttled, then this needs to be higher.